src/views/catalog/wizard/tabs/scripts/components/CloudInitNetworkForm.tsx

```diff
@@ -0,0 +1,199 @@
+  updateVM: UpdateValidatedVM;
+};
+
+const initialNetwork: CloudInitNetwork = {
+  network: {
+    version: '1',
+    config: [
+      {
+        type: 'physical',
+        name: '',
+        subnets: [
+          {
+            type: 'static',
+            address: [''],
+            gateway: '',
+          },
+        ],
+      },
+    ],
+  },
+};
+
+export const CloudinitNetworkForm: React.FC<CloudinitNetworkFormProps> = ({
+  cloudInitVolume,
+  updateVM,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  const [cloudinitConfigNetworkData, isBase64] = React.useMemo(
+    () => CloudInitDataHelper.getNetworkData(cloudInitVolume?.cloudInitNoCloud || {}),
+    [cloudInitVolume],
+  );
+
+  const [networkData, setNetworkData] = useImmer<CloudInitNetwork>(
+    typeof load(cloudinitConfigNetworkData) === 'object'
+      ? (load(cloudinitConfigNetworkData) as CloudInitNetwork)
+      : initialNetwork,
+  );
+
+  const onFieldChange = (field: string, value: string | string[]) => {
+    setNetworkData((draft) => {
+      ensurePath(draft, 'network.config');
+      if (!draft.network.config?.[0]) {
+        draft.network.config = [];
+      }
+
+      if (field === CloudInitNetworkFormKeys.NAME) {
+        draft.network.config[0].name = value as string;
+      } else {
+        if (!draft.network.config?.[0].subnets?.[0]) {
+          draft.network.config[0].subnets = [{ type: 'static', address: [''], gateway: '' }];
+        }
+        draft.network.config[0].subnets[0][field] = value as string;
+      }
+    });
+  };
+
+  const onUpdateVM = React.useCallback(
+    () =>
+      updateVM((vmToUpdate) =>
+        produceVMDisks(vmToUpdate, (vmDraft) => {
+          const cloudInitDiskName = cloudInitVolume?.name || 'cloudinitdisk';
+          const cloudInitDisk = vmDraft.spec.template.spec.domain.devices.disks.find(
+            (disk) => disk.name === cloudInitDiskName,
+          );
+
+          // cloudinitdisk deleted or doesn't exist, we need to re-create it
+          if (!cloudInitDisk) {
+            vmDraft.spec.template.spec.domain.devices.disks.push({
+              name: cloudInitDiskName,
+              disk: {
+                bus: 'virtio',
+              },
+            });
+          }
+
+          const updatedCloudinitVolume = {
+            ...(cloudInitVolume || { name: cloudInitDiskName }),
+            cloudInitNoCloud: {
+              ...(cloudInitVolume?.cloudInitNoCloud || {}),
+              ...CloudInitDataHelper.toCloudInitNoCloudNetworkSource(dump(networkData), isBase64),
+            },
+          };
+
+          const otherVolumes = getVolumes(vmDraft).filter((vol) => !vol.cloudInitNoCloud);
+          vmDraft.spec.template.spec.volumes = [...otherVolumes, updatedCloudinitVolume];
+        }),
+      ),
+    [cloudInitVolume, isBase64, updateVM, networkData],
+  );
+
+  React.useEffect(() => {
+    if (networkData) {
+      const ipAddreses = networkData.network.config[0].subnets[0].address.join(', ');
```

Comment:
ip adress are one string, we don't join or split it by `,` because we don't know what list format this has